translate/backend/csharp.py

```diff
@@ -0,0 +1,52 @@
+import ir
```

Comment:
Goodnesss you've removed all my notes and code that I'm going to have to add back in later.  For example, every function from the python module isn't necessarily a "test function".  That's why there are 2 separate visitors. And top-level python assignments will be handled differently than assignments inside functions.  Top-level python assignments will become fields in the test class an execution nodes code inside the TestInit function.  Why have you removed all that code? I'm just going to have to add it back in the next commit.

Reply:
> Goodnesss you've removed all my notes and code that I'm going to have to add back in later

Sorry, I probably should have described my intentions better in the PR. Given where I'm going with this and based on my next PR that I'm about to open, I don't think the deleted code needs to resurface. And if it did, it's in git history.

>  For example, every function from the python module isn't necessarily a "test function". That's why there are 2 separate visitors

I assume by this you mean the two `ir.IrVisitor` child classes that used to be in `csharpgen.py`? The IR shouldn't have two types of functions that are handled differently, that should be dealt with in the Python -> IR phase. In any case this PR is only about refactoring the C# code generator.